### PR TITLE
NEW: Added "Get lines and Post lines from BOM" at the REST Service

### PR DESCRIPTION
**Pull request by Humml87**
It´s my first small Pull Request in GIT->OpenSource. Please give me an feedback, if it´s necessary to optimize something.

### CODE REVIEW

File: htdocs/bom/class/api_boms.class.php [hunk 1]
```diff
@@ -279,6 +280,36 @@ public function delete($id)
...
+		if (!DolibarrApiAccess::$user->rights->bom->read) {
+			throw new RestException(401);
+		}
+
+		$result = $this->bom->fetch($id);
+		if (!$result) {
+			throw new RestException(404, 'BOM not found');
+		}
+
+		if (!DolibarrApi::_checkAccessToResource('bom_bom', $this->commande->id)) {
```
Comment (eldy) [inline htdocs/bom/class/api_boms.class.php:303]: Something seems wrong here: Do you mean 
$this->bom->id  and not $this->commande->id

**Comment by Humml87** [comment]
Hi, you right... Now it´s correct.
In the next step i like to integrate the POST, PUT, DELETE for the BOM REST Service.

**Comment by Humml87** [comment]
Hello @eldy, 
i found and read the "https://github.com/Dolibarr/dolibarr/blob/develop/.github/CONTRIBUTING.md".
My current PR is not matching the rules. I think the PR must come from a copy of my dolibarr fork. Also the brunch must come  from the developer-branch to the developer-branch.
It is usefull to close this PR and create a new with the correct rules? Or should this PR be solved differently?
Best regards Christian

P.S. I would like to apologize for the inconvenience.

**Comment by eldy** [comment]
Yes, your PR must be pushed into "develop" branch. But sometimes i can manually move this myself using a github feature to solve this (if there is no conflict when changing the branch). And i did it for your PR, so it is now correctly submitted for the develop branch. Si I succeed in merging it.
For next one, try to push it directly into "develop" when it is a new feature introduction

**Pull request merged**